app/routes.py

```diff
@@ -1,2 +1,183 @@
-from flask import Blueprint
+# Task List API Project
+# Katrina Kimzey
+# Cohort 15 - Paper
 
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, request, make_response, jsonify
+from flask_sqlalchemy import SQLAlchemy
+from sqlalchemy import asc, desc
+from datetime import datetime
+import os 
+from dotenv import load_dotenv
+
+tasks_bp = Blueprint("task", __name__, url_prefix="/tasks")
+
+load_dotenv()
+slack_path = 'https://slack.com/api/chat.postMessage'
+
+# ============== All Tasks ========================
+
+@tasks_bp.route("", methods=["GET", "POST"])
+def handle_tasks():
+    if request.method == "GET":
+        direction = request.args.get("sort")
+        if direction == "asc":
+            tasks = Task.query.order_by(asc("title"))
+        elif direction == "desc":
+            tasks = Task.query.order_by(desc("title"))
+        else:
+            tasks = Task.query.all()
+
+        tasks_response = [task.to_dict() for task in tasks]
+
+        return jsonify(tasks_response)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        try:
+            new_task = Task(title=request_body["title"],
+                            description=request_body["description"],
+                            completed_at=request_body["completed_at"]
+                            )
+        except KeyError:
+            return make_response({"details": "Invalid data"}, 400)
+
+        db.session.add(new_task)
+        db.session.commit()
+
+        return make_response({"task" : new_task.to_dict()}, 201)
+
+# ==================== Task by id ==============================
+
+@tasks_bp.route("/<active_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(active_id):
+    task = Task.query.get_or_404(active_id)
+
+    if request.method == "GET":
+        return {"task" : task.to_dict()}
```

Comment:
I have a few recommendations:
- Either using jsonify or make_response rather than just returning a dictionary. While Flask handles this gracefully, using jsonify or make_response (which explicitly states that it will convert dictionaries to JSON) adds an extra layer of error checking.
- Add status codes to all returns - this way it clear what your API is returning at each point, rather than using a default